### ISSUE TITLE
some question of rules : no-global-assign

### ISSUE THREAD

**Issue by rainfallman** [eslint/eslint#9789]
**Tell us about your environment**

* **ESLint Version:  v4.14.0**
* **Node Version: v8.9.0**
* **npm Version: v5.5.1**

**What parser (default, Babel-ESLint, etc.) are you using? default**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js



```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Here is my .eslintrc
```js
{
    "env": {
        "browser": true,
        "es6": true,
        "jquery":"true"
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ],
        "linebreak-style":[
            "error",
            "unix"
        ],
        "no-unused-vars":"off",
        "no-undef":"off",
        "no-console": 1,
        "no-unexpected-multiline":0,
        "no-global-assign":2
    }
}



```
And there is my js code
```js
const Map = {
    structure: [
        {
          ....
          .....

```

And someone tell me that it should show
```bash
Read-only global 'Map' should not be modified no-global-assign
```

**What did you expect to happen?**
It should told me error because i redefinition Map again,
but it didn't happen
Is there any wrong of my code?

**What actually happened? Please include the actual, raw output from ESLint.**

**Comment by platinumazure** [comment]
Hi @rainfallman, thanks for the issue.

The `no-global-assign` rule focuses on preventing the reassignment of globals: `Map = {};` It does not catch shadowing.

You should use the [`no-shadow` rule with `builtinGlobals: true` option](https://eslint.org/docs/rules/no-shadow#builtinglobals) to catch these cases.

**Comment by platinumazure** [comment]
Closing as the question has been answered.
